### PR TITLE
Removed normalization of arrows in 3D quiver

### PR DESCRIPTION
**Pull request by DanHickstein**
I removed the normalization of the arrows in 3D quiver to match the behavior of the 2D quiver plot. 

Also, I changed the default pivot point to be ‘tail’ in order to match the 2D quiver plot. 

Finally, clarified the comment about normalizing the ‘uvw’ variable.

This was suggested here:

https://github.com/matplotlib/matplotlib/issues/4211

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -2540,7 +2542,9 @@ def calc_arrow(uvw, angle=15):
         # arrow length ratio to the shaft length
         arrow_length_ratio = kwargs.pop('arrow_length_ratio', 0.3)
         # pivot point
-        pivot = kwargs.pop('pivot', 'tip')
+        pivot = kwargs.pop('pivot', 'tail')
+        # normalize
+        pivot = kwargs.pop('normalize', False)
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/axes3d.py]: uhm, I think you mean to assign to a variable called "normalize"

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -2601,7 +2605,10 @@ def calc_arrow(uvw, angle=15):
         # If any row of UVW is all zeros, don't make a quiver for it
         mask = norm > 1e-10
         XYZ = XYZ[mask]
-        UVW = UVW[mask] / norm[mask].reshape((-1, 1))
+        if normalize == True:
+            UVW = UVW[mask]/ norm[mask].reshape((-1, 1))
```
Comment (WeatherGod) [inline lib/mpl_toolkits/mplot3d/axes3d.py]: space before and after the `/`. Also, just do `if normalize:`.

**Comment by WeatherGod** [comment]
This will also likely need to update the image tests, I think (or at least have some of them use `normalize=True`).

**Comment by DanHickstein** [comment]
I fixed the typos. I don't know much about the tests, but I added "normalize=True" to the quiver3d_demo in order to keep it's behavior the same.

File: examples/mplot3d/quiver3d_demo.py [hunk 3]
```diff
@@ -14,6 +14,6 @@
 w = (np.sqrt(2.0 / 3.0) * np.cos(np.pi * x) * np.cos(np.pi * y) *
      np.sin(np.pi * z))
 
-ax.quiver(x, y, z, u, v, w, length=0.1)
+ax.quiver(x, y, z, u, v, w, length=0.1,normalize=True)
```
Comment (WeatherGod) [inline examples/mplot3d/quiver3d_demo.py]: need a space after the comma.

**Comment by WeatherGod** [comment]
The tests are located in `lib/mpl_toolkits/tests/test_mplot3d.py`

**Comment by tacaswell** [comment]
Sorry, I was confused about normalize vs the `angle` kwarg on `quiver` :sheep: 

Still think this is a good idea, but not for the reason I used to.

This needs an entry in https://github.com/matplotlib/matplotlib/tree/master/doc/api/api_changes with a note about how to return to the original behavior.

File: doc/api/api_changes/2015-11-12-DDH.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Modified the 3D quiver function in mpl_toolkits.mplot3d.axes3d.py to have the dafaults: normalize=False, pivot='tail'
+```````````````````````````
+Matplotlib has both a 2D and a 3D "quiver" function. These changes affect only the 3D function and make the default behavior of the 3D function match 2D version. There are two main changes:
+
+1) The 3D quiver function previously normalized the arrows to be the same length, which makes it unusable for situations where the arrows should be different lengths and does not match the behavior of the 2D function. This normalization behavior is now controlled with the "normalize" keyword, which defaults to False. 
+
+2) The "pivot" keyword now defaults to "tail" insteaf of "tip". This was done in order to match the default behavior of the 2D quiver function.
```
Comment (tacaswell) [inline doc/api/api_changes/2015-11-12-DDH.rst]: typo, sorry to be picky

File: doc/api/api_changes/2015-11-12-DDH.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+To obtain the previous behavior with the 3D quiver function, one can call the function with:
```
Comment (tacaswell) [inline doc/api/api_changes/2015-11-12-DDH.rst]: To work with rst the code examples should look like

```rst
last text line ::

   code example
```

Both the blank line and the 3 (!) spaces of indent matter

File: doc/api/api_changes/2015-11-12-DDH.rst [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+ax.quiver(x, y, z, u, v, w, normalize=True, pivot='tip')
+
+where "ax" is a axes3d object created with something like:
```
Comment (tacaswell) [inline doc/api/api_changes/2015-11-12-DDH.rst]: Use double backticks to get proper rst markup

**Comment by tacaswell** [comment]
If you don't normalize the vectors how does `length` come in?

**Comment by tacaswell** [comment]
In general :+1: .

**Comment by DanHickstein** [comment]
Thanks!
As far as I can tell, the length argument just scales the arrows. It's just a single number (nor an array), so it just scales them all together.

**Comment by DanHickstein** [comment]
I slightly modified the description of the length keyword argument to indicate that it doesn't actually set the length of the arrows when normalize is False (it just scales them). 

Let me know if there are other suggestions for this PR.

**Comment by tacaswell** [comment]
:+1: from me.  Leave this to @WeatherGod to push the button.

**Comment by WeatherGod** [comment]
Thank you @DanHickstein! @tacaswell, I take it that this will need to be cherry-picked to v2.0.x branch?

**Comment by tacaswell** [comment]
Yes

**Comment by WeatherGod** [comment]
cherrypicked to v2.0.x as bf6e9768ae

**Comment by DanHickstein** [comment]
Awesome! Thanks for your help!

**Comment by tacaswell** [comment]
@DanHickstein Thank you for your work!

**Comment by Zhuoliulz** [comment]
Feedback: Awsome! this modified file works properly and solves the "quiver" problems perfectly! Love your work!

**Pull request merged**